### ISSUE TITLE
[OTBN] Implement CSR access for base ISA

### ISSUE THREAD

**Issue by GregAC** [lowRISC/opentitan#3245]
> estimate 3

**Comment by GregAC** [comment]
Begun work on this but stopped for now as I don't think there's much point until we've resolved https://github.com/lowRISC/opentitan/issues/3338 and https://github.com/lowRISC/opentitan/issues/3350

The CSR accesses are either big number side registers or the random register so an implementation now has nothing to read/write from until we implement the big number side ALU (maybe RND could return some constant until we've got a proper random number source added).

**Comment by imphil** [comment]
I don't think we need to stall the base CSR instructions on figuring out the wide couterpart. The current code and simulators are fine with the normal RISC-V CSR behavior, so we can just implement the instructions this way and address potential spec changes in a follow-up.

**Comment by GregAC** [comment]
The RTL required to implement it is minimal and I was trying to come up with a consistent way to express CSR operations across base/bignum. How that all works depends upon how the bignum side works so I didn't think there was much point implementing it one way only to change it again.

As is a base side only implementations of the CSRs doesn't give you any new functionality  to work with as it won't be connected to anything until the wide side ALU exists (indeed it would appear to not work as you wouldn't actually write anything so a read back wouldn't give you the expected value) so lack of implementation shouldn't block much.

I think it makes more sense to implement CSR access once the initial wide side ALU is done.

**Comment by imphil** [comment]
I'm concerned that we're blocking on too much here. 

The current modexp.S (RSA) code works with CSRRS/CSRRW and BN.WSRRS/BN.WSRRW as specified in RISC-V, a behavior which is also implemented in both simulators we're  currently using. Implementing the same behavior in RTL will give us a well-understood starting point and shouldn't be too much wasted effort. Implementing CSR/WSR access is a requirement for MS3, spec clarifications/changes as discussed in #3338 and #3350 are not.

**Comment by GregAC** [comment]
modexp.S won't do much until the wide-side ALU is implemented anyway. So I'm saying I should get on with that first. I'll implement CSRs following that and if we're still undecided about CSR access I can implement the old method. I just think my current proposal of plain read/write instructions using the same encoding (make uses where you don't use the zero register illegal instructions) shouldn't be much effort to implement so if it's accepted I may find it's done by the time I've done the wide-side ALU.

**Comment by imphil** [comment]
Will be fixed as part of #3471.

**Comment by GregAC** [comment]
#3471 is merged so this is done
